pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,49 +381,59 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
-@register_infer_shape
+@register_useless
 @register_canonicalize
+@register_stabilize
 @register_specialize
 @node_rewriter([Subtensor])
 def local_replace_slice(fgraph, node):
     """
-    Remove Subtensor of the form:
-        1. X[0:-1:1] -> X[:-1]
-        2. X[0:-1] -> X[:-1]
-        3. X[:-1] -> X[:-1]
+    Rewrite Subtensor of the form:
+        1. X[0:-1:1] -> X[None:None:None]
```

Comment:
```suggestion
        1. X[0:-1:1] -> X[None:None:None]
```
:-1 is not the same as None, it will stop one entry before the end